clicmd

```diff
@@ -1,12 +1,68 @@
 #!/bin/bash -eu
 
+#
+# YADRO OpenBMC Command Line Interface
+# Copyright (C) 2020-2021 YADRO
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
 # Base CLI command handler.
 # This script is used for setting up the environment, elevating privileges
 # and handling hierarchical commands.
 
 # prevent modification of executable path
 PATH="/bin:/sbin:/usr/bin:/usr/sbin"
 
+# Role related definitions
+ROLE_ADMIN="admin"
+ROLE_OPERATOR="operator"
+ROLE_USER="user"
+
+GROUP_ADMIN="%GROUP_ADMIN%"
+GROUP_OPERATOR="%GROUP_OPERATOR%"
+GROUP_USER="%GROUP_USER%"
+
+# If we are running escalated, we have GROUP already set by the caller
+if [ "${USER}" != "root" -o ! -v GROUP ]; then
+  if [[ "${USER}" = "root" ]]; then
+    echo "Can't run CLI commands as root directly, use an admin account"
```

Comment:
Makes sense, will do. Didn't notice you output to `&2`, mea culpa.

Reply:
Also, is there any meaning behind `exit 2` after `"Invalid CLI command", should I change to `2` as well here?

Reply:
2 is ENOENT (No such file or directory).
I tried to use "correct codes" to notify a caller about specific error. It would be useful for tests or high level automation.
But I'm not sure if we need it now, do as you want.